staging/src/kubevirt.io/api/core/v1/types.go

```diff
@@ -1167,6 +1167,9 @@ type VirtualMachineSpec struct {
 	// FlavorMatcher references a flavor that is used to fill fields in Template
 	Flavor *FlavorMatcher `json:"flavor,omitempty" optional:"true"`
 
+	// FlavorSnapshot references a revision of flavor that is used to fill fields in Template
+	FlavorRevision *FlavorRevisionSpec `json:"flavorRevision,omitempty" optional:"true"`
```

Comment:
I'd expect this to go on the VM's Status and not the Spec. The difference here is that items on the Status section are not meant to be user modified. We don't want people to be able to alter the revision directly